airbyte-cdk/python/airbyte_cdk/sources/declarative/datetime/datetime_parser.py

```diff
@@ -37,6 +41,9 @@ def format(self, dt: datetime.datetime, format: str) -> str:
         # See https://stackoverflow.com/a/4974930
         if format == "%s":
             return str(int(dt.timestamp()))
+        if format == "%ms":
+            # Best we can do is second precision and mulitply by 1000
+            return str(int(dt.timestamp() * 1000))
```

Comment:
is a datetime object simply not expressive enough to retain millisecond value? looking at `timestamp()` sounds like it only returns seconds. But it just feels like a slight gotcha (although probably trivial) that we can receive milli timestamps from an API response, but end up truncating it when we save state. 

Just thinking aloud if we need to call this out in docs or something

Reply:
good question. Looks like my comment is wrong. timstamp does give ms precision, but the value is return as a float representing the number of seconds:


```
>>> d = datetime(1970, 1,1) + timedelta(milliseconds=16712345)
>>> d
datetime.datetime(1970, 1, 1, 4, 38, 32, 345000)
>>> d.timestamp()
45512.345
>>> d.timestamp() * 1000
45512345.0
```

Reply:
updated the test cases to show we retrain ms precision